cmd/minishift/cmd/start_preflight.go

```diff
@@ -0,0 +1,278 @@
+			"Checking if 'hyperv' driver configured",
+			false, startFlags.WarnCheckHyperVDriver.Name,
+			"Hyper-V virtual switch not set")
+	}
+}
+
+func checkXhyveDriver() bool {
+	_, err := exec.LookPath("docker-machine-driver-xhyve")
+	if err != nil {
+		return false
+	}
+	return true
+}
+
+func checkKvmDriver() bool {
+	_, err := exec.LookPath("docker-machine-driver-kvm")
+	if err != nil {
+		return false
+	}
+	return true
+}
+
+func checkHypervDriver() bool {
+	switchEnv := os.Getenv("HYPERV_VIRTUAL_SWITCH")
+	if switchEnv == "" {
+		return false
+	}
+	return true
+}
+
+func preflightChecksAfterStartHost(driver drivers.Driver) {
+	preflightCheckSucceedsOrFailsWithDriver(
+		startFlags.SkipInstanceIP.Name,
+		checkInstanceIP, driver,
+		"Checking for IP",
+		false, startFlags.WarnInstanceIP.Name,
+		"Error determining IP")
+	/*
+		// This happens too late in the preflight, as provisioning needs an IP already
+			preflightCheckSucceedsOrFailsWithDriver(
+				startFlags.SkipCheckNetworkHost.Name,
+				checkVMConnectivity, driver,
+				"Checking if VM is reachable from host",
+				startFlags.WarnCheckNetworkHost.Name,
+				"Please check our troubleshooting guide")
+	*/
+	preflightCheckSucceedsOrFailsWithDriver(
+		startFlags.SkipCheckNetworkPing.Name,
+		checkIPConnectivity, driver,
+		"Checking host reachable from VM",
+		true, startFlags.WarnCheckNetworkPing.Name,
+		"Instance is unable to ping outside")
+	preflightCheckSucceedsOrFailsWithDriver(
+		startFlags.SkipCheckNetworkDNS.Name,
+		checkDNSConnectivity, driver,
+		"Checking host reachable from VM",
+		true, startFlags.WarnCheckNetworkDNS.Name,
+		"Instance is unable to resolve address")
+	preflightCheckSucceedsOrFailsWithDriver(
+		startFlags.SkipCheckStorageMount.Name,
+		checkStorageMounted, driver,
+		"Checking storage disk mounted",
+		false, startFlags.WarnCheckStorageMount.Name,
+		"Instance did not mount storage")
+	preflightCheckSucceedsOrFailsWithDriver(
+		startFlags.SkipCheckStorageUsage.Name,
+		checkStorageUsage, driver,
+		"Checking storage disk usage",
+		false, startFlags.WarnCheckStorageUsage.Name,
+		"Instance does not have enough storage available")
+}
+
+func checkInstanceIP(driver drivers.Driver) bool {
+	ip, err := driver.GetIP()
+	if err == nil && net.ParseIP(ip).To4() != nil {
+		return true
+	}
+	return false
+}
+
+func checkVMConnectivity(driver drivers.Driver) bool {
+	// used to check if the host can reach the VM
+	ip, _ := driver.GetIP()
+
+	cmd := exec.Command("ping", "-n 1", ip)
+	stdoutStderr, err := cmd.CombinedOutput()
+	println(err.Error())
+	if err != nil {
+		return false
+	}
+	fmt.Printf("%s\n", stdoutStderr)
+	return false
+}
+
+func checkIPConnectivity(driver drivers.Driver) bool {
+	ipToPing := viper.GetString(startFlags.CheckNetworkIPHost.Name)
+	if ipToPing == "" {
+		ipToPing = "8.8.8.8"
+	}
+
+	fmt.Printf("\n   %s ... ", ipToPing)
+	return miniutil.IsIPReachable(driver, ipToPing, false)
+}
+
+func checkDNSConnectivity(driver drivers.Driver) bool {
```

Comment:
It needs to do a resolve first, and since the machine has been restarted any cache, is likely to be empty. I likely want can change it to a `curl` check, as this can probably also deal with proxy.